### PR TITLE
Improve Distribute Task

### PR DESCRIPTION
**Pull request by cristovaoth**
Includes:

- Separate distribute in two (prepare and apply). Make apply only run after verifying that checkpoints are persisted in the repo. It would be possible to propose things to a safe, to later realize that commit and push failed.
- Fix a bug in the Propose transaction: amountToFund in mainnet was incorrect
- Some code reorganization, including a simplification of what is written to schedule

### CODE REVIEW

File: tooling/src/tasks/distribute.ts [hunk 1]
```diff
@@ -1,41 +1,108 @@
...
+
+  const schedule = loadSchedule();
+  const preCount = checkpointCount();
+  const { merkleRootMainnet, merkleRootGnosis } = await hre.run("checkpoint", {
+    persist: false,
+  });
+  const postCount = checkpointCount();
+  // its important to ensure that checkpoint:apply is running
+  // on artifacts previously calculated and committed
+  assert(preCount === postCount, "Checkpoint task persist=false is writing?");
```
Comment (samepant) [inline tooling/src/tasks/distribute.ts:63]: this assert message is hard to parse, I'm not sure what it's saying

**Review by samepant — APPROVED** [review]

File: tooling/src/tasks/distribute.ts (hunk 1 above)
Comment (cristovaoth) [reply tooling/src/tasks/distribute.ts:63]: Right. So it's about ensuring that the checkpoint subtask calling during the apply pass is calculating the checkpoint but not persisting it (ensure that the persist:false is really taking effect). So it should have been priorly persisted. Its really coming from an abundance of caution, and preventing some scenario where a code refactoring changes this.

More context:
- distribute was split in prepare and apply
- prepare is to calculate and persist the checkpoint
- apply is to actually propose to safes
- if apply fails, and prepare doesn't, no biggie we have an extra checkpoint in the codebase no harm is done
- the reserve, i.e., if we post something to a safe that was not persisted, it's kind of a messy situation that will require manual resolution and will result in a broken claim frontend
